src/main/resources/application.yml

```diff
@@ -168,3 +173,10 @@ eq:
   
 report-settings:
   cron-expression: "0 * * * * *"
+
+cloud-storage:
+  case-schema-name: new-case
+  backoff-initial: 100
+  backoff-multiplier: 1.2
+  backoff-max: 3000
+  backoff-max-attempts: 2
```

Comment:
1) If CC is going to do Firestore retries then the back-off max and backoff-max-attempts need to be higher. I found that when reproducing this scenario in RH that Firestore could go awol for several seconds. Suggest just copying the values from RH or significantly increasing the max-attempts. See table in RH readme. 

2) Consider adding a summary of the retry code and its controlling properties. If not added then it'd be worth at least adding a short comment in the CC readme to point the reader to the Firestore retry section of the RH readme.

Reply:
I am happy for retry support to be left in - at least it is consistent with RH, even though it is highly unlikely to be needed. But then we didn't think we would need it in RH originally!
We have it, lets keep it. But agreed - if we are going to have it, then lets set the retries to a level that the mechanism will be of use should the need ever arise.

Reply:
Slightly different situation between Respondent Home and Contact Centre. Think it is easier for me to deal with the above points in reverse order.

RH is processing asynchronous requests originating from the message queue. It doesn't matter how long a backoff is used, it is a message queue waiting for success acknowledgement or failure. The RH readme describes the interaction between the retry mechanism on reading from the message queue, and retry on saving to Firestore and why we might want the two retry mechanisms on the same execution thread. I think referring to this readme would be confusing from CC. This simply uses the Spring @Retryable annotation on the method to store a case to Firestore. The configuration items for this annotation are well documented online. It is very little overhead on the amount of code, don't you love Spring, and far simpler than the RH scenario.

The CC request is a synchronous request from an operative waiting for a response. We want it to be as quick as possible, too long and they might just issue the request again, adding more load or just give up? In contrast to this don't want to issue lots of retries quickly that might be fruitless and just add to load for Firestore and CC. That is why I cut down the overall number of requests and time. Have set backoff-initial: 500, backoff-multiplier: 2, backoff-max: 3500, backoff-max-attempts: 3. If don't think suitable let me know what to set, can of course be overridden by environment variables anyway.

Reply:
I'd suggest backoff-initial=100, backoff-multiplier=1.2, backoff-max=6000 & backoff-max-attempts=14

Reply:
Phil discussed some values to use at present Peter. Is a different situation to RH where unlikely to have load problems but a retry still makes sense to me in case of any glitch, otherwise the client CC operator gets an error back. This may be raised again no doubt later down the line?